mitielib/mitie.py

```diff
@@ -555,4 +555,117 @@ def train(self):
 
+##############################################################################
+
+_f.mitie_create_text_categorizer_trainer.restype = ctypes.c_void_p
+_f.mitie_create_text_categorizer_trainer.argtypes = ctypes.c_char_p,
+
+_f.mitie_add_text_categorizer_labeled_text.restype = ctypes.c_int
+_f.mitie_add_text_categorizer_labeled_text.argtypes = ctypes.c_void_p, ctypes.c_void_p, ctypes.c_char_p
+
+_f.mitie_load_text_categorizer.restype = ctypes.c_void_p
+_f.mitie_load_text_categorizer.argtypes = ctypes.c_char_p,
+
+_f.mitie_save_text_categorizer.restype = ctypes.c_int
+_f.mitie_save_text_categorizer.argtypes = ctypes.c_void_p,
+
+_f.mitie_text_categorizer_trainer_get_beta.restype = ctypes.c_double
+_f.mitie_text_categorizer_trainer_get_beta.argtypes = ctypes.c_void_p,
+
+_f.mitie_text_categorizer_trainer_get_num_threads.restype = ctypes.c_ulong
+_f.mitie_text_categorizer_trainer_get_num_threads.argtypes = ctypes.c_void_p,
+
+_f.mitie_text_categorizer_trainer_set_beta.restype = None
+_f.mitie_text_categorizer_trainer_set_beta.argtypes = ctypes.c_void_p, ctypes.c_double
+
+_f.mitie_text_categorizer_trainer_set_num_threads.restype = None
+_f.mitie_text_categorizer_trainer_set_num_threads.argtypes = ctypes.c_void_p, ctypes.c_ulong
+
+_f.mitie_text_categorizer_trainer_size.restype = ctypes.c_ulong
+_f.mitie_text_categorizer_trainer_size.argtypes = ctypes.c_void_p,
+
+_f.mitie_train_text_categorizer.restype = ctypes.c_void_p
+_f.mitie_train_text_categorizer.argtypes = ctypes.c_void_p,
+
+_f.mitie_categorize_text.restype = ctypes.c_ulong
+_f.mitie_categorize_text.argtypes = ctypes.c_void_p, ctypes.c_void_p, ctypes.POINTER(ctypes.POINTER(ctypes.c_char_p)), ctypes.POINTER(ctypes.c_double) 
+
+
+class text_categorizer:
+    def __init__(self, filename):
+        self.__mitie_free = _f.mitie_free
+        if (isinstance(filename, ctypes.c_void_p)):
+            self.__obj = filename
+        else:
+            self.__obj = _f.mitie_load_text_categorizer(filename)
+        if (self.__obj == None):
+            raise Exception("Unable to load text_categorizer detector from " + filename)
+
+    def __del__(self):
+        self.__mitie_free(self.__obj)
+
+    def save_to_disk(self, filename):
+        """Save this object to disk.  You recall it from disk with the following Python
+        code: 
+            tcat = text_categorizer(filename)"""
+        if (_f.mitie_save_text_categorizer(filename, self.__obj) != 0):
+            raise Exception("Unable to save text_categorizer to the file " + filename);
+
+    
+    def __call__(self, tokens):
+        """Categorise a piece of text. The input should have been produced by 
+        tokenize().  This function returns a predicted label and a confidence score."""
+        score = ctypes.c_double()
+        label = ctypes.POINTER(ctypes.c_char_p)()
+        ctokens = python_to_mitie_str_array(tokens)
+        if (_f.mitie_categorize_text(self.__obj, ctokens, ctypes.byref(label), ctypes.byref(score)) != 0):
```

Comment:
Where is the memory allocated to label being freed?  It looks like it's leaked here.